core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -38,7 +38,14 @@
 .section .data
 .balign 4
 
-#ifdef CFG_BOOT_SYNC_CPU
+/*
+ * For VxWorks Support, the primary core will go through OP-TEE init
+ * without ever touching secondary cores; it is only after the primary
+ * core enters VxWorks, it issues special SMC to request booting of
+ * secondary cores. Thus CPU Sync is not required.
+ */
+
+#if defined (CFG_BOOT_SYNC_CPU) && !defined (CFG_TEE_VXWORKS_SUPPORT)
```

Comment:
> We want to reuse the existing plat-ls to support VxWorks

i would rather propose plat-stm and plat-ls to be changed:
```
-$(call force,CFG_BOOT_SYNC_CPU,y)
+CFG_BOOT_SYNC_CPU ?= y
```
To run VxWorks on a native optee/ls:
`make CFG_BOOT_SYNC_CPU=n`

Otherwise, in mk/config.mk
```
+ifeq ($(CFG_TEE_VXWORKS_SUPPORT),y)
+$(call force,CFG_BOOT_SYNC_CPU,n)
+endif
```

would that work?

My point is you intend to disable the full "boot sync cpu". So i would
rather have it disabled than adding extra check in the source code.


> It seems the generic boot process implementation is targeted for Firmware Code.
> The firmware would bring up all cores and put each through the OP-TEE;
>  VxWorks's usage differs from this:

no so. ATF support on aarch32 expects configuration sets explicitly
```
CFG_WITH_ARM_TRUSTED_FW=y
CFG_BOOT_SYNC_CPU=n
```
CFG_BOOT_SYNC_CPU is designed for that purpose.
I think the real issue is plat-ls (as plat-stm) forcing cpu sync at boot.